### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by am11**
Also cleaned up trailing whitespaces in the changeset files.

### CODE REVIEW

File: src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/AtomicComposition.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace System.ComponentModel.Composition.Hosting
     /// queryable state along with a sequence of actions necessary to complete the state when
     /// the atomicComposition is no longer in danger of being rolled back.  State is completed or
     /// rolled back when the atomicComposition is disposed, depending on the state of the
-    /// CompleteOnDipose property which defaults to false.  The using(...) pattern in C# is a
+    /// CompleteOnDispose property which defaults to false.  The using(...) pattern in C# is a
```
Comment (am11) [inline src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/AtomicComposition.cs:17]: There is not such property (even in the initial MEF1 port https://github.com/dotnet/runtime/commit/a6f350fabbcce17c84dc396dd744d9651ad1ca25).

**Comment by andchiind** [comment]
Another frequent typo is in System.Threading.Tasks.TaskCanceledException where cancelled is always spelled with one "L", including in error messages. An example instance is: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Threading.Tasks/ref/System.Threading.Tasks.Forwards.cs#L30

**Comment by am11** [comment]
> TaskCanceledException

This type name is a public contract. We don't change public contracts when fixing typos in bulk.

BTW, both spellings are [correct](https://www.merriam-webster.com/words-at-play/canceled-or-cancelled):

> The simple answer to the question of ‘is it canceled or cancelled’ is “either one is fine.”

and here are some quick stats which show that this repo favors single L spelling (though both are in use):

```sh
# count occurrences: 'cancelled' vs. 'canceled'

$ git grep -i cancelled | wc -l
     580
$ git grep -i canceled | wc -l
    4477
```

File: src/coreclr/inc/cordebug.idl [hunk 2]
```diff
@@ -6805,18 +6805,18 @@ interface ICorDebugArrayValue : ICorDebugHeapValue
                            length_is(cdim)] ULONG32 dims[]);
 
     /*
-     * HasBaseIndicies returns whether or not the array has base indices.
+     * HasBaseIndices returns whether or not the array has base indices.
      * If the answer is no, then all dimensions have a base index of 0.
      */
 
-    HRESULT HasBaseIndicies([out] BOOL *pbHasBaseIndicies);
+    HRESULT HasBaseIndices([out] BOOL *pbHasBaseIndices);
```
Comment (jkotas) [inline src/coreclr/inc/cordebug.idl:6812]: This is re-introducing the breaking change from https://github.com/dotnet/runtime/issues/76960

**Comment by am11** [comment]
CI failures are unrelated to changes:
* browser-wasm: https://github.com/dotnet/runtime/issues/78893
* win-x86: https://github.com/dotnet/runtime/issues/74838
* spmi replay is failing due to jitinterface GUID change in 720ae033fa8c8b9e933f198e03c4b16e8a432387 (pending new replays collection build)

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**